src/HavenoDaemon.test.ts

```diff
@@ -465,6 +465,102 @@ test("Can manage Monero daemon connections", async () => {
+  if (isMoneroNodeRunning) {
+    console.log("Warning: local Monero node is already running, skipping start and stop local Monero node test");
+  } else {
+    // expect error when passing in bad arguments
+    let badSettings = new MoneroNodeSettings();
+    badSettings.setStartupFlagsList(["--invalid-flag"]);
+    try {
+      await alice.startMoneroNode(badSettings);
+      throw new Error("should have thrown");
+    } catch (err) {
+      if (!err.message.startsWith("Failed to start monerod:")) throw new Error("Unexpected error: ");
+    }
+
+    // expect successful start with custom settings
+    let settings: MoneroNodeSettings = new MoneroNodeSettings();
+    let dataDir = rootDir + "/testdata";
+    let logFile = rootDir + "/testdata/test.log";
+    let p2pPort = 38080;
+    let rpcPort = 38081;
+    settings.setBlockchainPath(dataDir);
+    settings.setStartupFlagsList(["--log-file", logFile, "--p2p-bind-port", p2pPort.toString(), "--rpc-bind-port", rpcPort.toString(), "--no-zmq"]);
+    await alice.startMoneroNode(settings);
+    isMoneroNodeRunning = await alice.isMoneroNodeRunning();
+    assert(isMoneroNodeRunning);
+    
+    // expect settings are updated
+    let settingsAfter = await alice.getMoneroNodeSettings();
+    testMoneroNodeSettings(settings, settingsAfter!);
+
+    // expect connection to local monero node to succeed
+    let rpcUrl = "http://localhost:" + rpcPort.toString();
+    let daemon = await monerojs.connectToDaemonRpc(rpcUrl);
+    let height = await daemon.getHeight();
+    assert(height >= 0);
+
+    // expect stopped node
+    await alice.stopMoneroNode();
+    isMoneroNodeRunning = await alice.isMoneroNodeRunning();
+    assert(!isMoneroNodeRunning);
+    try {
+      daemon = await monerojs.connectToDaemonRpc(rpcUrl);
+      height = await daemon.getHeight();
+      throw new Error("should have thrown");
+    } catch (err) {
+      if (err.message !== "RequestError: Error: connect ECONNREFUSED 127.0.0.1:" + rpcPort.toString()) throw new Error("Unexpected error: " + err.message);
+    }
+
+    // start the node back up and continue test
+    await alice.startMoneroNode(settings);
+  }
+  
+  // expect error due to existing running node
+  let beforeSettings = await alice.getMoneroNodeSettings();
+  let newSettings = new MoneroNodeSettings();
+  newSettings.setStartupFlagsList(["--invalid-flag"]); 
+  try {
+    await alice.startMoneroNode(newSettings);
+    throw new Error("should have thrown");
+  } catch (err) {
+    if (err.message !== "Monero node already running") throw new Error("Unexpected error: " + err.message);
+  }
+
+  // expect settings are not modified if the start node fails
+  let settingsAfter = await alice.getMoneroNodeSettings();
+  if (beforeSettings === undefined) {
+    assert(settingsAfter === undefined); 
+  } else {
+    testMoneroNodeSettings(beforeSettings, settingsAfter!);
+  }
+
+  // remove local node connection if test generated one
```

Comment:
Should be unnecessary if the test does not generate a new local connection (the application will already hardcode local connections on 18081, 28081, 38081).